### PR TITLE
Add custom 404 page to website

### PR DESCRIPTION
**Pull request by a-mccarthy**
Signed-off-by: Abigail McCarthy <mabigail@vmware.com>

Adds a custom 404 page to the website. 

This also updates the default layout template (baseof.html) to make the 404 page possible.

### CODE REVIEW

File: site/layouts/404.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+<ul>
+  <li> <a href="/docs/latest">Documentation</a></li>
+  <li> <a href="/plugins">Plugins</a></li>
+  <li> <a href="/blog">Blog</a></li>
+  <li> <a href="/community">Community</a></li>
+  <li> <a href="/resources">Resources</a></li>
+</ul>
+
+<p> If you can't find what you are looking for, try asking a question in our <a href="https://kubernetes.slack.com/messages/velero">slack channel</a> or <a href="{{.Site.Params.gh_repo}}">filing an issue.</a><p>
```
Comment (carlisia) [inline site/layouts/404.html:19]: @a-mccarthy would you please make this link point to https://github.com/vmware-tanzu/velero/issues directly?

Comment (a-mccarthy) [reply site/layouts/404.html]: @carlisia updated! thanks for the review

**Review by zubron — COMMENTED** [review]
Thanks for this, @a-mccarthy! I have a question - is there a way to define a custom title for the 404 page? The current title for the page is "404 Page not found" (https://deploy-preview-3056--velero.netlify.app/doesnotexist) and I think it would be better to omit the "404" and have the title just be "Page not found".

**Comment by a-mccarthy** [comment]
@zubron Yes there is! i just pushed an update to remove the 404 from the title. All the wording on the page should be customizable :)

**Review by zubron — APPROVED** [review]
Thanks for making the title change! :)

**Review by ashish-amarnath — APPROVED** [review]

**Pull request merged**